pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -0,0 +1,381 @@
+	}
+
+	return false, nil
+}
+
+// CreateProcessNode finds or a create a new process activity node in the activity dump if the entry
+// matches the activity dump selector.
+func (at *ActivityTree) CreateProcessNode(entry *model.ProcessCacheEntry, generationType NodeGenerationType, shadowInsertion bool) (node *ProcessNode, newProcessNode bool, err error) {
+	// drop processes with abnormal paths
+	if entry.GetPathResolutionError() != "" {
+		return nil, false, fmt.Errorf("process cache entry error: %s", entry.GetPathResolutionError())
+	}
+
+	// look for a ProcessActivityNode by process cookie
+	if entry.Cookie > 0 {
+		var found bool
+		node, found = at.CookieToProcessNode[entry.Cookie]
+		if found {
+			return node, false, nil
+		}
+	}
+
+	defer func() {
+		// if a node was found, and if the entry has a valid cookie, insert a cookie shortcut
+		if entry.Cookie > 0 && node != nil {
+			at.CookieToProcessNode[entry.Cookie] = node
+		}
+	}()
+
+	// find or create a ProcessActivityNode for the parent of the input ProcessCacheEntry. If the parent is a fork entry,
+	// jump immediately to the next ancestor.
+	parentNode, newProcessNode, err := at.CreateProcessNode(entry.GetNextAncestorBinary(), Snapshot, shadowInsertion)
+	if err != nil || newProcessNode && shadowInsertion {
+		return parentNode, newProcessNode, err
+	}
+
+	// if parentNode is nil, the parent of the current node is out of tree (either because the parent is null, or it
+	// doesn't match the dump tags).
+	if parentNode == nil {
+
+		// since the parent of the current entry wasn't inserted, we need to know if the current entry needs to be inserted.
+		if !at.validator.MatchesSelector(entry) {
+			return nil, false, nil
+		}
+
+		// go through the root nodes and check if one of them matches the input ProcessCacheEntry:
+		for _, root := range at.ProcessNodes {
+			if root.Matches(&entry.Process, at.differentiateArgs) {
+				return root, false, nil
+			}
+		}
+
+		// we're about to add a root process node, make sure this root node passes the root node sanitizer
+		if !at.isValidRootNode(&entry.ProcessContext) {
+			if !shadowInsertion {
+				at.Stats.droppedCount[model.ExecEventType][invalidRootNodeReason].Inc()
+			}
+			return nil, false, fmt.Errorf("invalid root node")
+		}
+
+		// if it doesn't, create a new ProcessActivityNode for the input ProcessCacheEntry
+		node = NewProcessNode(entry, generationType)
+		if !shadowInsertion {
+			// insert in the list of root entries
+			at.ProcessNodes = append(at.ProcessNodes, node)
+			at.Stats.ProcessNodes++
+		}
+
+	} else {
+
+		// if parentNode wasn't nil, then (at least) the parent is part of the activity dump. This means that we need
+		// to add the current entry no matter if it matches the selector or not. Go through the root children of the
+		// parent node and check if one of them matches the input ProcessCacheEntry.
+		for _, child := range parentNode.Children {
+			if child.Matches(&entry.Process, at.differentiateArgs) {
```

Comment:
For both calls to Matches() (also on line 305):
Here the behavior changes from what we had before: we use to set this flags from config file for activity dumps, and force it to false for security profiles.
Making this generic would lead into some possible inconsistency: if some profiles are generated with DA yes (or no), then merged into a profile and send it back to the agent (or having an agent config change at some point), the loading agent could have DA no (or yes); and we will have a difference between the way we constructed the profile, and the way we handle it afterward. (sorry if I'm not clear enough here :'D)
tl;dr: I would rather prefer to keep the loose approach for profiles, by keeping this flags to false.
For the activity dump part, we could either keep it as it (choosing one of the two modes from the config) .. or, I wonder if we just can .. force it to false too. what did you think?